dm/config/subtask.go

```diff
@@ -274,6 +278,14 @@ func (c *SubTaskConfig) Adjust(verifyDecryptPassword bool) error {
 		return terror.ErrConfigOnlineSchemeNotSupport.Generate(c.OnlineDDLScheme)
 	}
 
+	if len(c.ShadowTableRule) == 0 {
```

Comment:
I think we should also add this adjust in `TaskConfig.adjust`. By the way, plase add following check to ensure that each regexp pattern contains exactly one submatch:
```
for _, r := range c.ShadowTableRule {
  p, err := regexp.Compile(r)
  if err != nil {
    ...
  }
  if p.NumSubexp() != 1 {
    return errors.New(...)
  }
}
```